etcdserver/server.go

```diff
@@ -862,3 +871,38 @@ func (s *EtcdServer) snapshot(snapi uint64, confState raftpb.ConfState) {
 func (s *EtcdServer) PauseSending() { s.r.pauseSending() }
 
 func (s *EtcdServer) ResumeSending() { s.r.resumeSending() }
+
+func (s *EtcdServer) ClusterVersion() *semver.Version {
+	s.verMu.Lock()
+	defer s.verMu.Unlock()
+	return s.clusterVersion
+}
+
+// monitorVersions checks the member's version every 10 seconds.
+// It updates the cluster version if all members agrees on a higher one.
+// It prints out log if there is a member with a higher version than the
+// local version.
```

Comment:
> do all members monitor all other members? Is this only ran by the leader? 

This is a so low cost thing (O(10 seconds) once). All members monitor all is just fine.

> why 10 seconds?

I think a O(10 second) is fine. A latency vs cost thing.

> Should we give the sysadmin a way to force this upgrade check?

Probably yes. Adding a endpoint to do this. But maybe in another pr?